pkg/instance/service.go

```diff
@@ -124,6 +156,8 @@ func (s service) validateNoCycles(instances []*model.DeploymentInstance) (graph.
 					return nil, fmt.Errorf("instance %q requires %q more than once", src.Name, requiredStack.Name)
 				} else if errors.Is(err, graph.ErrEdgeCreatesCycle) {
 					return nil, fmt.Errorf("link from instance %q to stack %q creates a cycle", src.Name, requiredStack.Name)
+				} else if errors.Is(err, graph.ErrVertexNotFound) {
+					return nil, fmt.Errorf("can't remove %q since at least one other instance depends on it: %s", requiredStack.Name, src.StackName)
```

Comment:
That error message only makes sense for the feature you are adding. You should phrase it in a way that makes sense for all features using this function.